modules/media/php/mediafileprovisioner.class.inc

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible files
+ * for the media menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\media;
+
+/**
+ * This class implements a data provisioner to get all possible files
+ * for the media menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class MediaFileProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create a MediaFileProvisioner, which gets files for the meida
+     * menu table.
+     */
+    function __construct()
+    {
+        parent::__construct(
+            "SELECT m.file_name,                                                   
+                    (SELECT PSCID from candidate WHERE CandID=s.CandID) as pscid,
```

Comment:
This sub query should not be needed as the table `candidate` is part of the join. 
```suggestion
                    c.PSCID as pscid,
```